.github/workflows/ci.yml

```diff
@@ -7,14 +7,53 @@ on:
   pull_request:
 
 jobs:
+  check-integrity:
+    name: Check integrity
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.20.7
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Check Go modules dependency file integrity
+        run: |
+          for module_file in $(find . -type f -name go.mod); do
+            module=$(dirname $module_file)
+            cd "$module"
+            go mod tidy
+            if [ "$(git status --porcelain)" != "" ]; then
+              printf >&2 '\n`go mod tidy` in module `%s` results in a dirty state, Go mod files are not in sync with the source code files, differences:\n\n%s\n\n' "$module" "$(git diff)"
+              git reset --hard
+              exit 1
+            fi
+            cd - > /dev/null
+          done
+
+      - name: Check generated file integrity
+        run: |
+          make generate manifests
+          git status --porcelain
+          git version
+          if [ "$(git status --porcelain)" != "" ]; then
+            printf >&2 '\n`make generate and manifests` results in a dirty state, generated files are not in sync with the source code files, differences:\n\n%s\n\n' "$(git diff)"
+            git reset --hard
+            exit 1
+          fi
+
   build:
+    name: Build
     runs-on: ubuntu-latest
 
     steps:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.19
+          go-version: 1.20.7
```

Comment:
**Question**: Why are we upgrading the running Go version here?
The module still uses 1.19, hence my question.

Obviously I mean the other place as well to be in sync with this.

Reply:
We have four Go modules in this repo, the script validates them all. Three of them use 1.19 but one is 1.20 (tests/e2e):
https://github.com/banzaicloud/koperator/blob/9c2ac523143facf437711698b4aa271ac03268a5/tests/e2e/go.mod#L3

Because of this, for `go mod tidy` to work we need `>=1.20`. Also I wanted to specify an exact version, not just `1.20`.

Reply:
The fourth one is being downgraded in a subsequent PR to 1.19 because it causes other problems as well.
For now I would suggest using 1.20  because of this.

Reply:
Thanks